image/docker_schema1.go

```diff
@@ -275,10 +274,14 @@ func (m *manifestSchema1) convertToManifestSchema2(uploadedLayerInfos []types.Bl
 		}
 
 		if !v1compat.ThrowAway {
+			layerDigest, err := digest.ParseDigest(m.FSLayers[v1Index].BlobSum)
+			if err != nil {
+				return nil, err
+			}
```

Comment:
@mtrmac No particular reason, transit from string to digest makes me uncertain, just a check for certain. Since we have make `BlobSum` a digest, just leave it to rest. What you said did bother me too when I replace/check all these digests.